frontend/src/components/templates/ILab/index.jsx

```diff
@@ -0,0 +1,241 @@
+
+import {
+  Accordion,
+  AccordionContent,
+  AccordionItem,
+  AccordionToggle,
+  Card,
+  CardBody,
+} from "@patternfly/react-core";
+import {
+  ExpandableRowContent,
+  Table,
+  Tbody,
+  Td,
+  Th,
+  Thead,
+  Tr,
+} from "@patternfly/react-table";
+import {
+  fetchILabJobs,
+  fetchMetricsInfo,
+  fetchPeriods,
+  setIlabDateFilter,
+} from "@/actions/ilabActions";
+import { formatDateTime, uid } from "@/utils/helper";
+import { useDispatch, useSelector } from "react-redux";
+import { useEffect, useState } from "react";
+import { useNavigate, useSearchParams } from "react-router-dom";
+
+import ILabGraph from "./ILabGraph";
+import MetaRow from "./MetaRow";
+import MetricsSelect from "./MetricsDropdown";
+import RenderPagination from "@/components/organisms/Pagination";
+import StatusCell from "./StatusCell";
+import TableFilter from "@/components/organisms/TableFilters";
+
+const ILab = () => {
+  const dispatch = useDispatch();
+  const navigate = useNavigate();
+  const [searchParams] = useSearchParams();
+
+  const { start_date, end_date } = useSelector((state) => state.ilab);
+  const [expandedResult, setExpandedResult] = useState([]);
+  const [expanded, setAccExpanded] = useState(["bordered-toggle1"]);
+
+  const onToggle = (id) => {
+    const index = expanded.indexOf(id);
+    const newExpanded =
+      index >= 0
+        ? [
+            ...expanded.slice(0, index),
+            ...expanded.slice(index + 1, expanded.length),
+          ]
+        : [...expanded, id];
+    setAccExpanded(newExpanded);
+  };
+  const isResultExpanded = (res) => expandedResult?.includes(res);
+  const setExpanded = async (run, isExpanding = true) => {
+    setExpandedResult((prevExpanded) => {
+      const otherExpandedRunNames = prevExpanded.filter((r) => r !== run.id);
+      return isExpanding
+        ? [...otherExpandedRunNames, run.id]
+        : otherExpandedRunNames;
+    });
+    if (isExpanding) {
+      dispatch(fetchPeriods(run.id));
+      dispatch(fetchMetricsInfo(run.id));
+    }
+  };
+
+  const { totalItems, page, perPage, tableData } = useSelector(
+    (state) => state.ilab
+  );
+
+  useEffect(() => {
+    if (searchParams.size > 0) {
+      // date filter is set apart
+      const startDate = searchParams.get("start_date");
+      const endDate = searchParams.get("end_date");
+
+      searchParams.delete("start_date");
+      searchParams.delete("end_date");
+      const params = Object.fromEntries(searchParams);
+      const obj = {};
+      for (const key in params) {
+        obj[key] = params[key].split(",");
+      }
+      dispatch(setIlabDateFilter(startDate, endDate, navigate));
+    }
+  }, []);
+
+  useEffect(() => {
+    dispatch(fetchILabJobs());
+  }, [dispatch]);
+
+  const columnNames = {
+    benchmark: "Benchmark",
+    email: "Email",
+    name: "Name",
+    source: "Source",
+    metric: "Metric",
+    begin_date: "Start Date",
+    end_date: "End Date",
+    status: "Status",
+  };
+
+  return (
```

Comment:
Is it possible for this thing to be split up? If so, do you think it would be worth it?